stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/WebSocketUtil.java

```diff
@@ -84,32 +87,39 @@ public static Annotation getServiceConfigAnnotation(Service service, String pkgP
     }
 
     public static void handleHandshake(WebSocketService wsService,
-                                       HttpHeaders headers, BStruct wsConnection, Context context,
+                                       HttpHeaders headers, WebSocketInitMessage initMessage, Context context,
                                        CallableUnitCallback callback) {
         String[] subProtocols = wsService.getNegotiableSubProtocols();
-        WebSocketInitMessage initMessage =
-                (WebSocketInitMessage) wsConnection.getNativeData(WebSocketConstants.WEBSOCKET_MESSAGE);
         int idleTimeoutInSeconds = wsService.getIdleTimeoutInSeconds();
         HandshakeFuture future = initMessage.handshake(subProtocols, true, idleTimeoutInSeconds * 1000, headers);
         future.setHandshakeListener(new HandshakeListener() {
             @Override
             public void onSuccess(Session session) {
-                BStruct serviceEndpoint = (BStruct) wsConnection.getNativeData(WebSocketConstants.WEBSOCKET_ENDPOINT);
-                populateEndpoint(session, serviceEndpoint);
-                wsConnection.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_SESSION, session);
+                // TODO: Need to create new struct
+                BStruct webSocketEndpoint = BLangConnectorSPIUtil.createBStruct(
+                        wsService.getResources()[0].getResourceInfo().getServiceInfo().getPackageInfo().getProgramFile(),
+                        PROTOCOL_PACKAGE_HTTP, WebSocketConstants.WEBSOCKET_ENDPOINT);
+                BStruct webSocketConnector = BLangConnectorSPIUtil.createBStruct(
```

Comment:
You can use the util method `createWebSocketConnectorBStruct( )` or you can delete this method if unused.
https://github.com/ballerina-platform/ballerina-lang/pull/6493/files#diff-905504cd527a475ef82691c3eb821853R63